### ISSUE TITLE
How works to PIN or not to PIN an optional bigfile video when streamed?

### ISSUE THREAD

**Issue by slrslr** [HelloZeroNet/ZeroNet#1484]
### Step 1: Please describe your environment

  * ZeroNet version: 0.6.3 rev3502
  * Operating system: W10
  * Web browser: FFbeta
  * Tor status: available
  * Opened port: no

### Step 2: Describe the problem:

Hello, it seems that on a zeronet based media streaming site the optional bigfile was PINned (making it undeletable by the "garbage collector") just because it was a few seconds played. The size on drive appears to be several hundred of megabytes even it was watched several seconds and closed as not worth to be watched/shared. Though the file is PINned :-/ [Here are](http://127.0.0.1:43110/Talk.ZeroNetwork.bit/?Topic:1530097903_1NZftAz9VSawGF5JgMYGFv4ByoKEaQbNZu#comment_168_1NZftAz9VSawGF5JgMYGFv4ByoKEaQbNZu) the details.

**Comment by HelloZeroNet** [comment]
>  The size on drive appears to be several hundred of megabytes even it was watched several seconds and closed as not worth to be watched/shared.

Currently we zerofill the file on first request, it could be possible to zerofill only until the current requested position. So if you streaming a file from the beginning it will only takes up the same ammount as the downloaded data, but if you seek into it, then it will zerofill the file until the requested position.

**Comment by slrslr** [comment]
@HelloZeroNet 

> if you streaming a file from the beginning it will only takes up the same ammount as the downloaded data

thank You for your attention, this seems not to be true in my case. I have not seeked in the video and it shows full size on my HDD. [Please take a look on the screenshot](http://127.0.0.1:43110/Me.ZeroNetwork.bit/?Post/1AbRRDveQD5EFtmXxBqQfS1guUWXxytMJZ/1L4dZcDF2maSKHDy788yhxpYnBWnXadUtS/1531290219).

**Comment by imachug** [comment]
Aah are you actually reading what was said?! @HelloZeroNet said that currently the file is filled with zeroes, and it *can* be possible not to do this.

**Comment by HelloZeroNet** [comment]
Yeah I already testing the modification and will roll out in the next days if no problems found with it.

**Comment by HelloZeroNet** [comment]
Space-saving bigfile preallocation added in Rev3535.

**Comment by slrslr** [comment]
Nice feature! I tried to play a webm video, and it worked good
Though when you try to load this video:
http://127.0.0.1:43110/big.kopykate.bit/?Video=1531883033_16NS3rBdW9zpLmLSQoD8nLTtNVsRFtVBhd
or
http://127.0.0.1:43110/big.kopykate.bit/?Video=1531884968_16NS3rBdW9zpLmLSQoD8nLTtNVsRFtVBhd
It **not started playing** in my case. The player shown just "loading" kind of animation. But on the HDD i see both mp4 files in ZeroNet\data\18Pfr2oswXvD352BbJvo59gZ3GbdbipSzh\data\users\16NS3rBdW9zpLmLSQoD8nLTtNVsRFtVBhd\ and always has the size:
5120 KB / 5,00 MB (5 242 880 bytes)
Same for m4v:
http://127.0.0.1:43110/big.kopykate.bit/?Video=1508189678_17ffrcHNzbFoUe81PyMEagQdv2DGNRLFcE
